File: lib/models/food.dart
import 'package:flutter/cupertino.dart';
/*
  Food Data Class
*/
class Food {
  final String name;
  final String designation;
  final int userFavorites;
  final int age;
  final String imgUrl;
  final String location;
  final String bio;
  bool isLiked;
  bool isFavorited;
  bool isDisliked;

  Food({
    required this.designation,
    required this.userFavorites,
    required this.name,
    required this.age,
    required this.imgUrl,
    required this.location,
    required this.bio,
    this.isLiked = false,
    this.isDisliked = false,
    this.isFavorited = false,
  });
}

File: react-app/packages/shared/src/models/food.ts
// Ported from: lib/models/food.dart

export interface Food {
  name: string;
  designation: string;
  userFavorites: number;
  age: number;
  imgUrl: string;
  location: string;
  bio: string;
  isLiked: boolean;
  isDisliked: boolean;
  isFavorited: boolean;
}

export function createDefaultFood(overrides: Partial<Food> & Pick<Food, 'name' | 'designation' | 'userFavorites' | 'age' | 'imgUrl' | 'location' | 'bio'>): Food {
  return {
    isLiked: false,
    isDisliked: false,
    isFavorited: false,
    ...overrides,
  };
}
